app/src/test/java/org/simple/clinic/settings/changelanguage/ChangeLanguageUpdateTest.kt

```diff
@@ -0,0 +1,96 @@
+package org.simple.clinic.settings.changelanguage
+
+import com.spotify.mobius.test.NextMatchers.hasEffects
+import com.spotify.mobius.test.NextMatchers.hasModel
+import com.spotify.mobius.test.NextMatchers.hasNoEffects
+import com.spotify.mobius.test.NextMatchers.hasNoModel
+import com.spotify.mobius.test.UpdateSpec
+import com.spotify.mobius.test.UpdateSpec.assertThatNext
+import junitparams.JUnitParamsRunner
+import junitparams.Parameters
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.simple.clinic.settings.Language
+
+@RunWith(JUnitParamsRunner::class)
+class ChangeLanguageUpdateTest {
+
+  private val defaultModel = ChangeLanguageModel.FETCHING_LANGUAGES
+
+  private val englishIndia = Language(displayName = "English", languageCode = "en_IN")
+  private val hindiIndia = Language(displayName = "हिंदी", languageCode = "hi_IN")
+
+  private val spec = UpdateSpec<ChangeLanguageModel, ChangeLanguageEvent, ChangeLanguageEffect>(ChangeLanguageLogic::update)
+
+  @Test
+  @Parameters(method = "params for initial models when loading selected language")
+  fun `when the selected language is loaded, the model must be updated`(model: ChangeLanguageModel) {
```

Comment:
Does this have to be a parameterized test?

Reply:
My intention was to show that regardless of whether the model was the empty model, or if the supported languages have already been loaded, it should update it with the current selected language. I didn't want to make two different tests because they were essentially testing the same thing.

But I agree with your later comment about coupling state transitions to the test. I'll break this into multiple tests.